ariesframework/src/main/java/org/hyperledger/ariesframework/credentials/CredentialService.kt

```diff
@@ -347,6 +348,22 @@ class CredentialService(val agent: Agent) {
         return CredentialAckMessage(credentialRecord.threadId, AckStatus.OK)
     }
 
+    /**
+     * Create an ``CredentialProblemReportMessage`` as response to a received offer.
+     *
+     * @param options options for the problem report message.
+     * @return credential problem report message.
+     */
+    suspend fun createProblemReport(options: AcceptOfferOptions): CredentialProblemReportMessage {
```

Comment:
> The function name looks general but handles only the offer message. It will be better to move this code to declineOffer().

I think it should be in the service as the following function is also in the service:
```
suspend fun createAck(options: AcceptCredentialOptions): CredentialAckMessage {
        var credentialRecord = credentialRepository.getById(options.credentialRecordId)
        credentialRecord.assertProtocolVersion("v1")
        credentialRecord.assertState(CredentialState.CredentialReceived)

        updateState(credentialRecord, CredentialState.Done)

        return CredentialAckMessage(credentialRecord.threadId, AckStatus.OK)
    }
```

I have changed the name of the function to `createOfferDeclinedProblemReport`

Reply:
I agree. It looks reasonable.